ui/components/app/snaps/snap-ui-selector/snap-ui-selector.tsx

```diff
@@ -118,15 +143,16 @@ export const SnapUISelector: React.FunctionComponent<SnapUISelectorProps> = ({
 
   const handleModalClose = () => setIsModalOpen(false);
 
-  const handleSelect = (value: string) => {
+  const handleSelect = (value: State) => {
     setSelectedOption(value);
+    onSelect?.(value);
     handleInputChange(name, value, form);
     handleModalClose();
   };
 
-  const selectedOptionIndex = options.findIndex(
-    (option) => option.value === selectedOptionValue,
-  );
+  const selectedOptionIndex =
+    findSelectedOptionIndex?.(selectedOptionValue) ??
```

Comment:
This allows to find the selected option if a simple string comparison is not enough